Extract ManageOptionController

JDawes-ScottLogic opened the pull request:
Extract ManageOptionController from ManageController and refactor front end to use it instead.
Added tests for refactor and new controller.

tpkelly commented on VotingApplication/VotingApplication.Web/Api/Models/DBViewModels/ManageOptionUpdateRequest.cs:
```diff
@@ -0,0 +1,21 @@
+
+    public class OptionUpdate
+    {
+        public string Name { get; set; }
```
We should mark Name as a required field

JDawes-ScottLogic commented:
Added Required field.
Investigating whether Except is needed. I added it later, so believe it is, but can't remember why (and the tests pass without it...).

Also found an issue with add/edit dialogs not working properly, which needs correcting.

tpkelly commented:
Looks fine to me, +1

stevenhillcox commented:
Let me know when you are happy for this to be merged.
